content/working_with/manager/update-deployment.md

```diff
@@ -248,66 +232,39 @@ please use the `--auto-correct-types` flag along with `cfy deployments update`.
-```yaml
-# original deployment blueprint
-node_templates:
-    node1:
-        relationships:
-          - type: cloudify.relationships.contained_in
-            target: node2
-```
-```yaml
-# deployment update blueprint
-node_templates:
-    node1:
-        relationships:
-          - type: cloudify.relationships.contained_in
-            target: node3  # unsupported update - can't modify a contained_in relationship's target
-```
-### Relationship Properties
-You cannot change a relationship's property, for example, `connection_type`.
-```yaml
-# original deployment blueprint
-node_templates:
-    node1:
-        relationships:
-          - [...]
-            properties:
-                connection_type: all_to_all
-```
-```yaml
-# deployment update blueprint
-node_templates:
-    node1:
-        relationships:
-          - [...]
-            properties:
-                connection_type: all_to_one  # unsupported update - can't modify a relationship's property
-```
-### Groups, Policy Types and Policy Triggers
-You cannot make changes in the top level fields `groups`, `policy_types` and `policy_triggers` as a part of a deployment update blueprint.
+To avoid reinstalling instances when their properties have changed, implement the `check_drift` and `update` operations. Instances of nodes that implement those operations, will run them instead of being reinstalled.
+
+### The `check_drift` operation
+
+The `cloudify.interfaces.lifecycle.check_drift` operation should examine the node properties, instance runtime properties, and the actual state of the provisioned resource, and compute the differences.
+
+The instance will be considered drifted if this operation returns a non-empty value (for example, a Python dict describing the differences).
+
+The {{< param cfy_manager_name >}} doesn't inspect the returned value, so it can be any object. Plugin authors are advised to return a description of all the differences, so that the `update` operation can act upon them.
+
+{{% note title="Warning" %}}
+If `check_drift` returns an empty or false value, `update` operations will not run, and even in case of blueprint changes (eg. if the node properties changed), the instances will not be updated or reinstalled. Take care to always return a non-empty value if there are _any_ changes to the instances.
+{{% /note %}}
+
+If the `check_drift` operation is not implemented, the instances are only considered drifted if there are relevant blueprint changes (eg. the node properties changed).
+
+### The `update` operations
+
+The `update` operations are:
+- `cloudify.interfaces.lifecycle.update`
+- `cloudify.interfaces.lifecycle.update_config`
+- `cloudify.interfaces.lifecycle.update_apply`
+
+Plugin authors can implement any, or all, of those operations. Those three operations can be written in a way to mirror the `create`, `configure`, and `start` operations.
```

Comment:
```suggestion
Plugin authors can implement any, or all, of those operations. Those three operations can be written in a way so that they mirror the `create`, `configure`, and `start` operations.
```